community/boilerplates/auth-jwt-roles/README.md

```diff
@@ -0,0 +1,208 @@
+
+The endpoints to manage users are very limited (it is only possible to create a new user through the `/signup` endpoint). This is kind of a choice as this service is meant to be used for authentication only. The user and roles management can be done through the Hasura Graphql Engine or any other service accessing to the same database.
+
+## Rationale
+
+See this [issue](https://github.com/hasura/graphql-engine/issues/1420).
+
+## Database schema
+
+Three tables are used:
+
+- `user`:
+  - `id`: UUID. Primary key. Automatically generated.
+  - `username`: String. Unique user identifier.
+  - `password`: String. Hashed with bcrypt.
+  - `active`: Boolean. If not active, not possible to connect with this user.
+- `role`:
+  - `id`: UUID. Primary key. Automatically generated.
+  - `name`: String. Unique role identifier.
+- `user_role`:
+  - `id`: UUID. Primary key. Automatically generated.
+  - `role_id`: UUID. Foreign key that references the `id` of the `role` table.
+  - `user_id`: UUID. Foreign key that references the `id` of the `user` table.
+
+## Prerequisites
+
+- PostgreSQL
+- Node.js 8.9+
+
+## Getting Started
+
+### Environment variables
+
+_Note: you can find examples of RSA keys in the repository. **DO NOT USE THEM FOR PRODUCTION!**_
+
+- `AUTH_PRIVATE_KEY="-----BEGIN RSA PRIVATE KEY-----\nypPTIfSzZ399o........"`
+
+  RSA private key used to sign the JWT. You need to escape the lines with "\n" in the variable. If the variable is not set, it will try to use the private.pem file.
+
+- `AUTH_PUBLIC_KEY="-----BEGIN PUBLIC KEY-----\nV02/4RJi........"`
+
+  RSA private key used to deliver the JWK set. You need to escape the lines with "\n" in the variable. Please not that this feature is not working yet. If the variable is not set, it will try to use the public.pem file.
+
+- `AUTH_KEY_ID="<unique-id-for-this-key>"`
+
+  Used to identify the key currently used to sign the tokens. If the variable is not set, a hash string will be generated from the public key and used instead.
+
+- `DATABASE_URL=postgres://<username>:<password>@<host>:<port>/<database_name>`
+
+  URL to connect to the Postgres database. The format is . For instance: `DATABASE_URL=postgres://postgres:@localhost:5432/postgres`
+
+- `PORT=8080`
+
+The port the server will listen to.
+
+### Build and deploy on Docker (production)
+
+First you need to build the image and to tag it:
+
+```bash
+docker build . -t platyplus/authentication:latest
+```
+
+TODO: document on how to deploy on docker.
+
+You can also have a look at [this docker-compose gist](https://gist.github.com/plmercereau/b8503c869ffa2b5d4e42dc9137b56ae1) to see how I use this service in a docker stack with Hasura and [Traefik](https://traefik.io/).
+
+### Deploy locally (developpment)
+
+```bash
+# Clone the repo
+git clone https://github.com/platyplus/authentication-server
```

Comment:
Can we change the git clone and cd instructions to reflect the Hasura repo?

Reply:
@shahidhk I think also we can add this into `community/tools/auth-servers/passportjs-jwt-roles` because user can use it with Hasura side by side and it's not just a boilerplate :thinking: 
Are you agree guys?

**/CC:** @plmercereau

Reply:
thanks guys for your feedback. @shahidhk I have just made the changes accordingly

Reply:
@mnlbox Tools are for tools that can be used with Hasura, like json2graphql etc. I don't think this will fit in there.

Reply:
@shahidhk agreed, that's why I used the directory you suggested